### ISSUE TITLE
WSL2 and Ryzen Master - Virtualization Based Security.

### ISSUE THREAD

**Issue by rwilliams** [microsoft/WSL#4771]
* Your Windows build number:
19041.1 - Slow Ring

* What you're doing and what's happening:  Attempting to run Ryzen Master with WSL2 enabled
![image](https://user-images.githubusercontent.com/29144/71111320-07485c00-217e-11ea-9f37-d47d2336326f.png)

* What's wrong / what should be happening instead:
This is the expected behavior since updates to Ryzen Master now require VBS off. 

* Workarounds

1. I've found the [DG Readiness Tool](https://www.microsoft.com/en-us/download/details.aspx?id=53337) via reddit which allows disabling VBS but it doesn't persist through reboots. Would it be possible to disable VBS persistently? 
2. Is uninstalling and re-installing Virtual Machine Platform as needed better work around? I did notice my existing distro wouldn't boot after re-installing Virtual Machine Platform until I loaded another distro from the Windows Store. wsl --shutdown also just hangs. It seems like installing a new distro re-initializes tome things broken from adding and removing Virtual Machine Platform. *It might just take a 2nd reboot.

* Real Life Security Implications
What are the security implications of disabling VBS? 

 * Background
Ryzen Master is probably the best tool for managing Ryzen CPU performance. Because of its VBS disabled requirement it becomes useless for anyone using Hyper-v, Docker, WSL, Virtual Machine Platform, etc.

**Dyanamic Ryzen Master Patcher**
https://github.com/milesburchell/RyzenMasterVBSFix thanks @milesburchell

**Current Offsets**
With the offset you change `0F 84` to `90 E9`

Build 2.2.0.1543
**3000 Series** 

- 3900x/3950x - 0x0163E7

**2000 Series** 

**1000 Series** 

Build 2.1.1.1472
**3000 Series** 

- 3900x/3950x - 0x0158C7
- 3800x/3700x(?) - 0x158C0 

**2000 Series** 
- 0x0125E9

**1000 Series** 
- 1200/1900X - 0x000C5A2.

**Comment by onomatopellan** [comment]
Try disabling Memory integrity. 
https://techcommunity.microsoft.com/t5/Windows-Insider-Program/Windows-Defender-System-Guard-Making-a-leap-forward-in-platform/m-p/167303

**Comment by rwilliams** [comment]
@onomatopellan Thanks for that link. Memory Integrity is disabled.

**Comment by onomatopellan** [comment]
VBS is not the problem then. WSL2 enables an Hyper-V component and it seems Ryzen Master v13 won't work with Hyper-v enabled. You can read more here https://community.amd.com/thread/241792

**Comment by rwilliams** [comment]
@onomatopellan When you reboot after using the DG Readiness Tool you are prompted with this
![IMG_20191218_144631](https://user-images.githubusercontent.com/29144/71130014-c2d0b680-21a5-11ea-80a3-867afd120999.jpg)
Disabling that allows Ryzen master to work along side WSL2, but it doesn't not persist between reboots.

**Comment by MostHated** [comment]
I have not seen that come up, so I have not been able to test it. Is there any confirmation or this, or a way to force that dialog to show so I can test it?

**Comment by MostHated** [comment]
Hmmm... nevermind, I just tried again and now it came up. I disabled it and WSL2 is still working, and I was able to open Ryzen Master. Thanks for that, I spent hours on AMD forums and other places trying to find something that would work, even trying the Readiness tool before but it didn't seem to work before.

**Comment by Stanzilla** [comment]
You can HEX edit RyzenMaster.exe to remove the check for that as well and it works fine.

**Comment by rwilliams** [comment]
Interesting. If it works fine with VBS enabled why are they doing the check in the first place.

**Comment by Stanzilla** [comment]
That's a question neither their Twitter, nor their Forum and Email support could answer me. All I got was

```
The feedback from our software team is that the incompatibility lies with VBS limiting Ryzen Master access to specific system resources.

We are currently looking into this limitation, but do not have any specific timeline as to when a solution will be available.

Thanks for contacting AMD. 
```

**Comment by rwilliams** [comment]
@Stanzilla Can you email the detail on the hex edit? My email is on my profile if you click it.

**Comment by Stanzilla** [comment]
I can just link this Tweet by @tomrus88 https://twitter.com/TOM_RUS/status/1204867886197755904

**Comment by yunghoy** [comment]
This was why I uninstalled WSL2 and installed Linux on my external ssd. The only reason I was on the fast ring instead of 19h1 was for WSL2 but it seemed useless because my cpu was not that great without Ryzen Master.

**Comment by rwilliams** [comment]
I did the same thing, put Ubuntu on a spare SSD as it was less hassle than rebooting to disable VBS or WSL.

**Comment by onomatopellan** [comment]
Another succesfull story on how to run Ryzen Master with WSL2:
https://twitter.com/steeve/status/1222282378648924163

**Comment by rwilliams** [comment]
Has anyone using it with the hex edit seen any issues with Ryzen Master? It's like AMD is checking for it but it doesn't actually effect the software?

**Comment by rwilliams** [comment]
@benhillis Can we get an official response to this? There are a lot of people in this boat and it's pretty frustrating at the moment.

**Comment by Stanzilla** [comment]
It's AMD's fault not Microsoft's. And yeah, I've been running a HEX edited version just fine over the last few months.

**Comment by rwilliams** [comment]
I understand it's not Microsoft's fault but maybe MS and AMD could work together and figure out why this is even a thing?

**Comment by orangpelupa** [comment]
anyone have the hex edit for the latest ryzen master? the hex offset has changed :(

**Comment by AntiPhotonz** [comment]
For version 2.1.1.1472 the offset 0x0158C0

**Comment by tautomer** [comment]
> For version 2.1.1.1472 the offset is 0x0158C0.

Can confirm it works with 2.1.1.1472.

**Comment by codewerfer** [comment]
> For version 2.1.1.1472 the offset is 0x0158C0.

To keep it exact, it is at `0x0158C7`.

**Comment by skythian** [comment]
My Hex at that offset is different... This is 2.1.1.1472
![Annotation 2020-02-24 092332](https://user-images.githubusercontent.com/44532899/75170492-6cbd4980-56e7-11ea-9b06-e4e38af98cfc.jpg)

**Comment by tommyvct** [comment]
Same. I even tried to find the previous and after bytes in the file but nothing was found. 
They must have changed the memory map without out knowledge.

**Comment by codewerfer** [comment]
I've downloaded it again an reinstalled it. Seams to stay at version 2.1.1.1472:
Signed at `27.01.2020 09:13:41` by `Advanced Micro Devices INC.` (fingerprint `ef03f44c216eaf43950f7f7f13c683885904f46d`)
Filesize: `817920 Bytes` (798 KiB)
CRC32: `0EE45B94`

Installed (and downloaded) for Ryzen 9 3950X.

**Comment by Stanzilla** [comment]
It should have 0F 84 at 0x0158c7 and you change that to 90 E9, worked fine here with Master 2.1.1.1472

**Comment by AntiPhotonz** [comment]
Searching for `0F84F7000000` might work if the address of the jump hasn't changed. Then it's just a matter of trying all returned results for the one that works.

**Comment by tommyvct** [comment]
> It should have 0F 84 at 0x0158c7 and you change that to 90 E9, worked fine here with Master 2.1.1.1472

Nope. 
![image](https://user-images.githubusercontent.com/7903172/75597351-974a3200-5a5a-11ea-823e-fa6989db7dda.png)



> Searching for `0F84F7000000` might work if the address of the jump hasn't changed. Then it's just a matter of trying all returned results for the one that works.

Nope.
![image](https://user-images.githubusercontent.com/7903172/75597399-edb77080-5a5a-11ea-8ce0-f7605d6c3599.png)

**Comment by tommyvct** [comment]
> Installed (and downloaded) for Ryzen 9 3950X - makes the only reason for my why there is a difference for the same "version".

I have a Ryzen 5 1600.

**Comment by StevenPG** [comment]
@Stanzilla Do you know how to determine the location? I wouldn't mind throwing a site up that has each hex value set for each version (and try to automate it to some degree).

**Comment by gavin19** [comment]
> I have a Ryzen 5 1600.

I get the same `74 24` as you, and also a Gen 1 (1800x). File size is also different (854KB), and it was signed 24/9/19.

I'm still using a pre 1.3 build which doesn't need any special tweaks to run. Are the newer versions even worth using for Gen 1 anyway? The only difference I can see is that the newer versions have the SMT toggle.

**Comment by tautomer** [comment]
It's `0x0158C0` for my Ryzen Master version `2.1.1.1472` and I have a 3800X. So Ryzen Master binary is different for different CPU's?

**Comment by gabriel-peracio** [comment]
My Ryzen master version is `2.1.1.1472` and `AMD Ryzen Master.exe` has a SHA-1 of `9470C71DB450FAD40B4972AD8B22F10B3086174E`.

I could not find anything appropriate at `0x0158c7` so here's what I did:

1. Download IDA freeware
2. Open `AMD Ryzen Master.exe`
3. hit `ctrl+L` (or `jump`>`jump by name`)
4. select `aVirtualization` in the window that opens
5. on the window that opens, right click the blue `aVirtualization`  (it's the second "column") and select `Jump to xref to operand...`. You should now see a graph visualization of the code. This is the code that invokes the nag message box (since it references the string contained in that message)
6. Double click the arrows going into said box. This will take you to whatever code invoked the one you're currently looking at. Keep going "backwards" until you find a box that looks like the one in the tweet (https://twitter.com/steeve/status/1222277573624975366/photo/1). It ends with a `jz` to some offset.
7. Click the `jz`. On the info bar at the bottom of the window, you will see the address. Mine was `000125E9`.
8. Armed with that offset, you can now open whatever your hex editor is and go to that offset. You should see `0F 84`. Change that to `90 E9`.
9. Profit

**Comment by WestCruX** [comment]
> Tl;DR: I have a 2700x and my offset was `0x0125E9`. Seems like the binaries are different.
> 
> My Ryzen master version is `2.1.1.1472` and `AMD Ryzen Master.exe` has a SHA-1 of `9470C71DB450FAD40B4972AD8B22F10B3086174E`.
> 
> I could not find anything appropriate at `0x0158c7` so here's what I did:
> 
> 1. Download IDA freeware
> 2. Open `AMD Ryzen Master.exe` (it will ask you for a bunch of things. I don't know what difference they make. The defaults were fine for me.)
> 3. hit `ctrl+L` (or `jump`>`jump by name`)
> 4. select `aVirtualization` in the window that opens
> 5. on the window that opens, right click the blue `aVirtualization`  (it's the second "column") and select `Jump to xref to operand...`. You should now see a graph visualization of the code. This is the code that invokes the nag message box (since it references the string contained in that message)
> 6. Double click the arrows going into said box (do not make the mistake I did and click the arrows going away from the box, this is the code that displays the nag message, there's no point to going there). This will take you to whatever code invoked the one you're currently looking at. Keep going "backwards" until you find a box that looks like the one in the tweet (https://twitter.com/steeve/status/1222277573624975366/photo/1). It ends with a `jz` to some offset.
> 7. Click the `jz`. On the info bar at the bottom of the window, you will see the address. Mine was `000125E9`.
> 8. Armed with that offset, you can now open whatever your hex editor is and go to that offset. You should see `0F 84`. Change that to `90 E9`.
> 9. Profit
> 
> I know nothing about IDA and this was my first time debugging like this. I'm glad I had that tweet with the screenshots so I could at least see if I was getting closer!
> 
> This was fun 😅

Can you make a little video to explaine how to edit? From the fifth point i'm lost 😅 thanks a lot in advance!

**Comment by gabriel-peracio** [comment]
Maybe the following images help:

Steps 3 and 4:
![ida1](https://user-images.githubusercontent.com/8999086/75791435-db944580-5d6c-11ea-96a0-28c56aa1d029.png)

Step 5:
![ida2](https://user-images.githubusercontent.com/8999086/75791445-dfc06300-5d6c-11ea-97aa-64298c89f06d.png)

Steps 5 and 6 (after confirming your selection, you should arrive here. Notice the current selected node has a blue highlight at the top. Double clicking the arrows leading into that box focuses on whatever box leads to that one. You can also visually follow the arrows if you prefer):
![ida3](https://user-images.githubusercontent.com/8999086/75791454-e18a2680-5d6c-11ea-9e88-e75920b65b4b.png)

Step 7 (You can reach this box fairly quickly, by double clicking the red arrow at the top of the image above, then just copy the address IDA gives you):
![ida4](https://user-images.githubusercontent.com/8999086/75791464-e3ec8080-5d6c-11ea-8528-a1d318816661.png)

**Comment by tommyvct** [comment]
@gabriel-peracio Hell year it worked!!!!! Thank you so much!!!!

**Comment by rwilliams** [comment]
So can we pin down how many different versions there are of the executable per release? I've got a 3950x and used 0x0158C0 i believe. Also I can update the main post with the current offsets per version if folks can provide them for 2000's and 1000's cpu's.

**Comment by Stanzilla** [comment]
3900x and 0x0158c7

**Comment by rwilliams** [comment]
I'll try and keep the main post updated as offsets get posted.

**Comment by tautomer** [comment]
3800x and 0x150C0. I guess it's related to generations and aslo die configuration, since 3900x/3950x have different offset than 3800x. So the offset for 3700x should also be 0x158C0.

**Comment by skythian** [comment]
@gabriel-peracio Amazing. Nice job! For Threadripper 1900X the offset is 0x00c5a2.

**Comment by TehWiz** [comment]
Ryzen 1200 is at 0x000C5A2

**Comment by Xeno1177** [comment]
Used 0x0158C7 for 3900x and confirmed the offset with IDA using the instructions above.  When running Ryzen Master, the splash screen displays then the system reboots.  Any thoughts on what I might have done wrong?  Running Master 2.1.1.1472

**Comment by thomasczer** [comment]
https://github.com/microsoft/WSL/issues/4771#issuecomment-593188388
Can confirm this worked on my 3600 too. Thanks!

**Comment by WestCruX** [comment]
> Maybe the following images help:
> 
> Steps 3 and 4:
> ![ida1](https://user-images.githubusercontent.com/8999086/75791435-db944580-5d6c-11ea-96a0-28c56aa1d029.png)
> 
> Step 5:
> ![ida2](https://user-images.githubusercontent.com/8999086/75791445-dfc06300-5d6c-11ea-97aa-64298c89f06d.png)
> 
> Steps 5 and 6 (after confirming your selection, you should arrive here. Notice the current selected node has a blue highlight at the top. Double clicking the arrows leading into that box focuses on whatever box leads to that one. You can also visually follow the arrows if you prefer):
> ![ida3](https://user-images.githubusercontent.com/8999086/75791454-e18a2680-5d6c-11ea-9e88-e75920b65b4b.png)
> 
> Step 7 (You can reach this box fairly quickly, by double clicking the red arrow at the top of the image above, then just copy the address IDA gives you):
> ![ida4](https://user-images.githubusercontent.com/8999086/75791464-e3ec8080-5d6c-11ea-8528-a1d318816661.png)

Thank you very much! Y've saved my Ryzen Master!

**Comment by stevefan1999-personal** [comment]
> 
> 
> Tl;DR: I have a 2700x and my offset was `0x0125E9`. Seems like the binaries are different.
> 
> My Ryzen master version is `2.1.1.1472` and `AMD Ryzen Master.exe` has a SHA-1 of `9470C71DB450FAD40B4972AD8B22F10B3086174E`.
> 
> I could not find anything appropriate at `0x0158c7` so here's what I did:
> 
>     1. Download IDA freeware
> 
>     2. Open `AMD Ryzen Master.exe` (it will ask you for a bunch of things. I don't know what difference they make. The defaults were fine for me. You probably have to run IDA as admin so that it is able to edit the file in `C:/Program Files/`)
> 
>     3. hit `ctrl+L` (or `jump`>`jump by name`)
> 
>     4. select `aVirtualization` in the window that opens
> 
>     5. on the window that opens, right click the blue `aVirtualization`  (it's the second "column") and select `Jump to xref to operand...`. You should now see a graph visualization of the code. This is the code that invokes the nag message box (since it references the string contained in that message)
> 
>     6. Double click the arrows going into said box (do not make the mistake I did and click the arrows going away from the box, this is the code that displays the nag message, there's no point to going there). This will take you to whatever code invoked the one you're currently looking at. Keep going "backwards" until you find a box that looks like the one in the tweet (https://twitter.com/steeve/status/1222277573624975366/photo/1). It ends with a `jz` to some offset.
> 
>     7. Click the `jz`. On the info bar at the bottom of the window, you will see the address. Mine was `000125E9`.
> 
>     8. Armed with that offset, you can now open whatever your hex editor is and go to that offset. You should see `0F 84`. Change that to `90 E9`.
> 
>     9. Profit
> 
> 
> I know nothing about IDA and this was my first time debugging like this. I'm glad I had that tweet with the screenshots so I could at least see if I was getting closer!
> 
> This was fun 😅

I might have an alternative way to do this.

So if you decompile the function you can see much clearer what that jump is doing.

![](https://i.imgur.com/3LHhtw9.png)

And sub_14003E5E0 is can actually be called "CheckVirtualizationBasedSecurityStatus", according to one snippet in the decompiled function:

![](https://i.imgur.com/P0ftfNl.png)

Let's go back to the original function, then we can clearly deduce that "CheckVirtualizationBasedSecurityStatus" is an out function, and we passed something probably called so that the result returns whether the system is running under hypervisor or not.

![](https://i.imgur.com/4TYrEoB.png)

Let's go back to the assembly, you can see that all is making sense:

![](https://i.imgur.com/TMNsPxZ.png)

Thus, we can either patch the jump, or we can actually patch the function call. 

The signature to the call I found out is "E8 ? ? ? ? 44 39 6D A8". This is not guaranteed for future major updates if the execution path/optimization is changed.

We can then patch the call with nops. Very vile but it should work.

**Comment by Stanzilla** [comment]
There is a a new version, 2.2.0.1543, which still does not work with VBS enabled.

**Comment by AntiPhotonz** [comment]
For my ryzen 3900x, the offset for version 2.2.0.1543 is 0x0163E7.

**Comment by Allaen** [comment]
> For my ryzen 3900x, the offset for version 2.2.0.1543 is 0x0163E7.

Can you tell which new value we should enter at the offset? Thanks

**Comment by rwilliams** [comment]
> Armed with that offset, you can now open whatever your hex editor is and go to that offset. You should see 0F 84. Change that to 90 E9.

**Comment by rwilliams** [comment]
> For my ryzen 3900x, the offset for version 2.2.0.1543 is 0x0163E7.

Works for 3950x as well.

**Comment by stevefan1999-personal** [comment]
Well, regarding that I have also made a little Go script that automatically patches the Ryzen Master to work with WSL2 -- Despite now I'm really sick of Golang now because it is so annoying:
```golang
package main

import (
	"bufio"
	"errors"
	"fmt"
	"io"
	"os"
	"path"
	"time"

	"rsc.io/binaryregexp"
)

type PatchUnitExecutableInfo struct {
	dirPath string
	name    string
}

type PatchUnit struct {
	regex      *binaryregexp.Regexp
	executable PatchUnitExecutableInfo
}

var ryzenMaster = PatchUnit{
	binaryregexp.MustCompile("(\u00E8.{4})\u0044\u0039\u006D\u00A8"),
	PatchUnitExecutableInfo{path.Join(os.Getenv("ProgramW6432"), "AMD", "RyzenMaster", "bin"), "AMD Ryzen Master.exe"},
}

func patchRyzenMaster() error {
	joinDirPath := func(name string) string {
		return path.Join(ryzenMaster.executable.dirPath, name)
	}

	if fp, err := os.OpenFile(joinDirPath(ryzenMaster.executable.name), os.O_RDWR, 0666); err != nil {
		return err
	} else {
		defer fp.Close()
		createBackup := func() error {
			t := time.Now()
			formatted := fmt.Sprintf("%d-%02d-%02dT%02d-%02d-%02d",
				t.Year(), t.Month(), t.Day(),
				t.Hour(), t.Minute(), t.Second())
			fp.Seek(0, io.SeekStart)
			if fpBackup, err := os.Create(joinDirPath(fmt.Sprintf("%s-%s.bak", ryzenMaster.executable.name, formatted))); err != nil {
				return err
			} else {
				defer fpBackup.Close()
				_, err := io.Copy(fpBackup, fp)
				return err
			}
		}

		reader := bufio.NewReader(fp)
		i, err := fp.Seek(0, io.SeekCurrent)
		if err != nil {
			return err
		}
		strMatch := ryzenMaster.regex.FindReaderSubmatchIndex(reader)
		if len(strMatch) <= 0 {
			return errors.New("can't find the signature")
		}
		fmt.Printf("[+] Found the signature from Ryzen Master at 0x%X!", strMatch[2])
		if err := createBackup(); err != nil {
			return err
		}
		fmt.Println("[+] Backup of Ryzen Master created!")
		if _, err = fp.Seek(i, io.SeekStart); err != nil {
			return err
		}
		if _, err = fp.WriteAt([]byte{0x90, 0x90, 0x90, 0x90, 0x90}, int64(strMatch[2])); err != nil {
			return err
		}
		fmt.Printf("Written NOPs to 0x%X!\n", strMatch[2])
	}
	return nil
}

func main() {
	if err := patchRyzenMaster(); err != nil {
		panic(err)
	} else {
		fmt.Println("Ryzen Master is now patched to work with WSL2!")
	}
}
```
I hope in the future I don't have to work with Golang, this is really really long and annoying and verbose as hell. (well, there are much more worse language such as Python)

**Comment by Allaen** [comment]
> [#4771 (comment)](https://github.com/microsoft/WSL/issues/4771#issuecomment-593633375)
> 
> > Armed with that offset, you can now open whatever your hex editor is and go to that offset. You should see 0F 84. Change that to 90 E9.

Thanks, I was being stupid, tried to change the offset in the installer...

**Comment by z-h-z-h** [comment]
here short video how to patch ryzen master https://www.youtube.com/watch?v=58KapiWEcs4

**Comment by milesburchell** [comment]
To future thread finders, I wrote a helper program that dynamically patches Ryzen Master at launch instead of editing the binary. This should be robust across updates and does not invalidate the code signing of the executable on disk.

https://github.com/milesburchell/RyzenMasterVBSFix

**Comment by AriESQ** [comment]
My problem is that this is not just Ryzen Master. Critical tools I rely on, CPU-Z, HWINFO64, OCCT also don't work with Hyper-V enabled. I suspect this is because with Hyper-V enabled, the "Host" operating system is also running under Hyper-V albeit with additional hardware privileges.

It's interesting to me that we're able to patch Ryzen Master. Which tells me that it is checking for VBS, but VBS isn't preventing it from working. AMD may be cautious about instability or features they wish to add in the future.

**Comment by Stanzilla** [comment]
> CPU-Z, HWINFO64, OCCT also don't work with Hyper-V enabled.

Cannot confirm that, those work fine here.

**Comment by tommyvct** [comment]
Now even VMWare can coexist with Hyper-V, why Ryzen Master can't?

**Comment by mattlacey** [comment]
Clone @milesburchell's repo, built it, tried to run it from WSL just for kicks (which failed), launched it from cmd and it worked a charm. Good work!

**Comment by alminisl** [comment]
> 
> 
> Clone @milesburchell's repo, built it, tried to run it from WSL just for kicks (which failed), launched it from cmd and it worked a charm. Good work!

Stupid question, but how do I run a .sln file from CMD ?

**Comment by rwilliams** [comment]
The `.sln` file is a Visual Studio solution file not the application.

**Comment by alminisl** [comment]
> 
> 
> The `.sln` file is a Visual Studio solution file not the application. On the release page there is a link the executable [milesburchell/RyzenMasterVBSFix/releases](https://github.com/milesburchell/RyzenMasterVBSFix/releases)

hah.. I'm blind.. thanks :D

**Comment by mattlacey** [comment]
@alminisl I must be blind too, because I downloaded and build it because I didn't see the release!

**Comment by NeuronButter** [comment]
> well you are stupid you it seems or jealous because someone else made better way to fix this then you

You weren't the one who made that. You just copied someone else's work from Github, made a quite rushed video on you doing literally what the docs says, and shamelessly published the link, wanting views and likes. Don't defend someone else's code, don't defend your shameless crying for YouTube views, and certainly don't come back to Github when you have no knowledge of what actually is going on. The text on the video is the default for Premiere, literally all of what you've said here is grammatically incorrect, you joined Github this month, you've had no contribution whatsoever to the actual computer science or programming behind all of this, you probably have pirated Premiere Pro because you don't work yet, so if I didn't know any better, you're 10, turning 11 in 9 months.

Enjoy your dislike, and go play Roblox, because at your age, it's the only remotely productive thing you can do, and don't come into a Github thread with the toxicity and shamelessness like yourself. Come back when you've turned at least 13 so you're old enough by Github's ToS. Good day.

**Comment by stevefan1999-personal** [comment]
@NeuronButter the moment I realized this bitch ass @Nerdistnoob can't code, I refused to continue arguing

**Comment by tommyvct** [comment]
@stevefan1999-personal can't code and make noob videos doesn't mean you should calling his name using bad languages. 
Be friendly.

**Comment by stevefan1999-personal** [comment]
@tommyvct I just told him to fuck off because 1. his video is cringy, and 2. that still requires patching which is explicitly stated in his cringe shit. btw thanks to you i forgot to dislike his video.

**Comment by gabriel-peracio** [comment]
I must remind everyone that this thread is starting to violate the contributing guidelines and code of conduct of both Github and Microsoft. I personally don't care, at some point someone with auth will swoop in and delete a load of comments, but on the other hand we risk having this thread locked, which would be a shame.

This is far from "fixed" (although I think Microsoft is completely blameless here), but has a very neat workaround available for now, and we have some people coming here from google to solve their ryzen master woes (especially developers, since if you try to use wsl+docker+ryzen master, AMD's official instructions basically amount to "you're screwed").

The thread is "idle" because there is little we can do besides
- keep patching ryzen master
- keep bugging AMD for an official fix, since the VBS prompt is completely useless (as we've proven, otherwise it wouldn't work even after being patched)

**Comment by FragsAreUs** [comment]
I know this may be a old thread but for people with the latest version of Ryzen master 2.3.0.1591 the offset is located at 00019CC0 0A / 0B and had to change them also from 0F 04 to 90 E9

**Comment by rwilliams** [comment]
@FragsAreUs  are you using https://github.com/milesburchell/RyzenMasterVBSFix ?

**Comment by FragsAreUs** [comment]
> @FragsAreUs are you using https://github.com/milesburchell/RyzenMasterVBSFix ?

woops was not paying attention that this was the microsoft github and not for ryzen master but i was using this

https://github.com/klauspost/ryzen-master-vbs-patch

but did not work for me. My bad

**Comment by stumins** [comment]
@milesburchell thanks for the awesome tool!

Worked flawlessly for me on a Ryzen 3900x, Windows build 19041.450 with SVM enabled in BIOS, WSL2 booting fine, etc.

**Comment by ivanzud** [comment]
Thanks @milesburchell ! Your tool also worked for me.

**Comment by tdimarzio** [comment]
Thanks @milesburchell ! Excellent work and greatly appreciated.

**Comment by Ritesh294** [comment]
I've noticed that whenever SVM is enabled in the BIOS, Windows reports the maximum performance percentage as 100 for all cores. With SVM disabled Windows will report calculated maximum performance percentages for each core. The performance percentages can be found in Event Viewer -> Window Logs -> System -> Event 55 Kernel-Processor-Power (Each core will have its own event).
Here's an example of what I found for my Ryzen 5 3600:
SVM DISABLED:
Core 1 - Maximum performance percentage: 123
           - Minimum performance percentage: 100
Core 2 - Maximum performance percentage: 116
           - Minimum performance percentage: 61
Core 3 - Maximum performance percentage: 130
           - Minimum performance percentage: 61
Core 4 - Maximum performance percentage: 127
           - Minimum performance percentage: 61
SVM ENABLED
Core 1 - Maximum performance percentage: 100
           - Minimum performance percentage: 100
Core 2 - Maximum performance percentage: 100
           - Minimum performance percentage: 100
Core 3 - Maximum performance percentage: 100
           - Minimum performance percentage: 100
Core 4 - Maximum performance percentage: 100
           - Minimum performance percentage: 100

**Comment by tyscorp** [comment]
I was able to install and run Ryzen Master 2.6.0.1702 with SVM enabled and no patches on my 3700X with AGESA 1.0.0.6. (not sure if that makes a difference)

**Comment by stijndp** [comment]
Can confirm. SVM is enabled and 2.6.0.1702 of Ryzen Master can open without trickery.

To update Ryzen Master if you can't open Ryzen Master; run \AMD\AutoUpdate\AMDAutoUpdate.exe 001
After the update, it should open even with SVM enabled in BIOS.

**Comment by rwilliams** [comment]
I'm going to close this as it appears AMD is playing nice now. I'll re-open if it comes up again. Thank you all for contributing.

**Comment by numpad0** [comment]
Don't seem to work for me. Could it be AGESA or A/B testing on AMD side?

![image](https://user-images.githubusercontent.com/4281428/104742782-37519c80-578e-11eb-85b3-0a9cdf39f98d.png)

- 10 Pro 10.0.19041.685
- Ryzen 5 2600X
- ASRock Fatal1ty B450 Gaming K4
- AGESA Combo-AM4 1.0.0.1 (according to HWiNFO64)
- Ryzen Master 2.6.0.1702 (according to AMD\AutoUpdate\VersionInfo.xml )

- WSL2, Hyper-V enabled
- no particular trickery

**Comment by WaffelBit** [comment]
Updated Ryzen Master to 2.6.0.1702 and it now works.
But I do **NOT** have Hyper-V activated, just the VirtualMachinePlatform

- Win 10 Education 20H2 19042.746
- Ryzen 3 3300X
- ASRock B550M Pro4 
- AGESA Combo-AM4 v2 1.0.0.2

**Comment by bovcan** [comment]
well, now it looks it will work. version 2.6.0.1xxx works with HyperVM enabled

**Comment by cmfcmf** [comment]
> Don't seem to work for me. Could it be AGESA or A/B testing on AMD side?

I had the same problem, same versioninfo.xml file. It worked after uninstalling and reinstalling instead of just upgrading Ryzen Master.

**Comment by DGulshan** [comment]
> > Don't seem to work for me. Could it be AGESA or A/B testing on AMD side?
> 
> I had the same problem, same versioninfo.xml file. It worked after uninstalling and reinstalling instead of just upgrading Ryzen Master.

It didn't work for me either way, upgrading OR uninstalling and reinstalling, both still give the VBS error.

**Comment by Fredyy90** [comment]
Finally working for my again, running hyper-v and active VMs in the background.

Ryzen Master 2.6.1.1797
Ryzen 3900X
AsRock x470 Taichi (BIOS L3.71)
AGESA Combo-AM4 1.0.0.4

**Comment by tsc2017** [comment]
I am using 1920x and Gigabyte x399 Gaming 7.
Ryzen Master 2.6.1.1797 still doesn't work for me. But the `VBSFix.exe` is good enough.

**Comment by Avellea** [comment]
I apologize about necroing this issue... Has there been any updates in the last year or so? VBS/Hyper-V enabled on Win11 with Ryzen Master 2.11.2.2633 does not open, nor does VBSFix work...

Is the best course of action to downgrade somehow?

**Comment by lowestprime** [comment]
Not at all, I had the same problem.

This worked for me (Windows 11 Education Version 23H2 OS build 22631.4037; AMD Ryzen 5 2400G; AMD Radeon RX 580; WSL Version 2.2.4.0 Distribution Ubuntu-24.04; Ryzen Master 2.11.2.2633):
![image](https://github.com/user-attachments/assets/e3f87a13-85a9-46ab-9cff-71565b340794)

Briefly:
1. Download and unzip [dgreadiness_v3.6.zip](https://download.microsoft.com/download/B/D/8/BD821B1F-05F2-4A7E-AA03-DF6C4F687B07/dgreadiness_v3.6.zip); [dgreadiness_v3.6.zip](https://github.com/user-attachments/files/16692983/dgreadiness_v3.6.zip)
2. Navigate to the unzipped dgreadiness_v3.6 directory containing DG_Readiness_Tool_v3.6.ps1 in admin PowerShell.
3. Run:
   1. Set-ExecutionPolicy unrestricted
   2. .\DG_Readiness_Tool_v3.6.ps1 -Disable
4. Reboot and follow instructions pressing F3 to confirm all changes.
5. It may be necessary to repeat these steps if the changes revert after reboot.

Source:
[Permanently Disable Virtualization-Based Security (VBS) in Windows 10 Pro](https://superuser.com/questions/1511884/permanently-disable-virtualization-based-security-vbs-in-windows-10-pro#comment2768426_1511884)
[VMware Workstation and Device/Credential Guard are not compatible](https://stackoverflow.com/a/53442486/6646109)
